modules/effects/arrow-effects-reactor-data/src/main/kotlin/arrow/effects/reactor/FluxK.kt

```diff
@@ -24,6 +24,12 @@ data class FluxK<A>(val flux: Flux<A>) : FluxKOf<A>, FluxKKindedJ<A> {
   fun <B> map(f: (A) -> B): FluxK<B> =
     flux.map(f).k()
 
+  fun <B> mapFilter(f: (A) -> Option<B>): FluxK<B> =
+    flux
+      .filter { f(it).isDefined() }
+      .map { f(it).orNull()!! }
```

Comment:
.flatMap { f(it).fold({ Flux.empty() }, { Flux.just(it) }) }

Reply:
Same for ObservableK too.

Reply:
Really really better!